commandapi-annotations-reloaded/src/main/java/dev/jorel/commandapi/annotations/reloaded/annotations/Subcommands.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright 2018, 2020 Jorel Ali (Skepter) - MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package dev.jorel.commandapi.annotations.reloaded.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * When Subcommand is used multiple times on the same element, this contains all of them.
+ */
+@Retention(RetentionPolicy.SOURCE)
+@Target( { ElementType.METHOD, ElementType.TYPE })
+public @interface Subcommands {
```

Comment:
This is actually a feature of how Java annotation processors work. You don't annotate `@Subcommands` directly on anything, instead, this allows multiple uses of `@Subcommand` and internally it will place them in this array for you to access in order. See [Repeating Annotations](https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html)

Reply:
This is neat. I did not know this, all I saw was an annotation that takes an array of another annotation and was like "Huh?"